Jenkinsfile

```diff
@@ -1,128 +1,141 @@
+              def jacocoExcludes =
+                      // build tools
+                      "**/org/eclipse/jetty/ant/**" +
+                              ",**/org/eclipse/jetty/maven/**" +
+                              ",**/org/eclipse/jetty/jspc/**" +
+                              // example code / documentation
+                              ",**/org/eclipse/jetty/embedded/**" +
+                              ",**/org/eclipse/jetty/asyncrest/**" +
+                              ",**/org/eclipse/jetty/demo/**" +
+                              // special environments / late integrations
+                              ",**/org/eclipse/jetty/gcloud/**" +
+                              ",**/org/eclipse/jetty/infinispan/**" +
+                              ",**/org/eclipse/jetty/osgi/**" +
+                              ",**/org/eclipse/jetty/spring/**" +
+                              ",**/org/eclipse/jetty/http/spi/**" +
+                              // test classes
+                              ",**/org/eclipse/jetty/tests/**" +
+                              ",**/org/eclipse/jetty/test/**";
+              step([$class: 'JacocoPublisher',
+                    inclusionPattern: '**/org/eclipse/jetty/**/*.class',
+                    exclusionPattern: jacocoExcludes,
+                    execPattern: '**/target/jacoco.exec',
+                    classPattern: '**/target/classes',
+                    sourcePattern: '**/src/main/java'])
+              // Report on Maven and Javadoc warnings
+              step([$class: 'WarningsPublisher',
+                    consoleParsers: [
+                            [parserName: 'Maven'],
+                            [parserName: 'JavaDoc'],
+                            [parserName: 'JavaC']
+                    ]])
+            }
+            if(isUnstable())
+            {
+              notifyBuild("Unstable / Test Errors", jdk)
+            }
+          }
         }
+      } catch(Exception e) {
+        notifyBuild("Test Failure", jdk)
+        throw e
       }
-    }
-  } catch(Exception e) {
-    notifyBuild("Test Failure")
-    throw e
-  }
 
-  try
-  {
-    stage 'Compact3'
+      try
+      {
+        stage ('Compact3') {
 
-    dir("aggregates/jetty-all-compact3") {
-      withEnv(mvnEnv) {
-        sh "mvn -B -Pcompact3 clean install"
+          dir("aggregates/jetty-all-compact3") {
+            withEnv(mvnEnv) {
+              sh "mvn -V -B -Pcompact3 clean install"
+            }
+          }
+        }
+      } catch(Exception e) {
+        notifyBuild("Compact3 Failure", jdk)
+        throw e
       }
     }
-  } catch(Exception e) {
-    notifyBuild("Compact3 Failure")
-    throw e
   }
 }
 
+
 // True if this build is part of the "active" branches
 // for Jetty.
 def isActiveBranch()
 {
   def branchName = "${env.BRANCH_NAME}"
   return ( branchName == "master" ||
-           branchName.startsWith("jetty-") );
+          branchName.startsWith("jetty-") );
```

Comment:
can you make this a regex instead? `jetty-.*\.x` ?
All branches that start with `jetty-` and ending with `.x` ?

Reply:
Regex isn't needed, it can be `( (branchName.startsWith("jetty-") && branchName.endsWith(".x") )`

Reply:
this will not pick up branch name such ' jetty-9.4.x-2164-update-poms-serviceloader-with-osgi '

Reply:
done